license-bill-of-materials.go

```diff
@@ -488,7 +491,12 @@ func longestCommonPrefix(licenses []License) string {
 			break
 		}
 		for _, c := range n.Children {
-			prefix = append(prefix, c.Name)
+			if c.Shared > 1  {
```

Comment:
i think we still have issue in this case.

if two pkg shares more than one part, then we might end up with skipping the parts in the middle.

for example, /a/b/c/d/e and /a/b/c/d/e/f/g's common prefix will be /a/ in this case. Since the node `b`, `c`...'s share is greater than 1 and we will skip them. am i misunderstanding the algo here?